### ISSUE TITLE
`MissingPrimaryKey` false positive when primary key lives in a `Spark.Dsl.Fragment`

### ISSUE THREAD

**Issue by jeroen11dijk** [leonqadirie/ash_credo#133]
**Check:** `AshCredo.Check.Warning.MissingPrimaryKey`
**ash_credo version:** 0.12.1

## Summary

`MissingPrimaryKey` flags resources whose primary key is defined inside a `Spark.Dsl.Fragment` module rather than the resource module itself. The fragment is composed into the resource at compile time via `use Ash.Resource, fragments: [...]`, so the resource has a valid PK at runtime — but the check only inspects the AST of the resource source file and never opens the fragment.

This is particularly common for composite primary keys built from `belongs_to` relationships in join tables.

## Reproduction

`lib/my_app/messaging/resources/quick_message_response.ex`:

```elixir
defmodule MyApp.Messaging.QuickMessageResponse do
  @moduledoc false
  use Ash.Resource,
    fragments: [
      MyApp.Messaging.QuickMessageResponse.Fragments.QuickMessageResponseRelationships
    ],
    data_layer: AshPostgres.DataLayer,
    domain: MyApp.Messaging

  postgres do
    table "quick_message_responses"
    repo MyApp.Repo
  end

  actions do
    defaults [:read, :destroy, create: :*, update: :*]
  end

  multitenancy do
    strategy :attribute
    attribute :company_id
  end
end
```

`lib/my_app/messaging/resources/fragments/quick_message_response_relationships.ex`:

```elixir
defmodule MyApp.Messaging.QuickMessageResponse.Fragments.QuickMessageResponseRelationships do
  @moduledoc false
  use Spark.Dsl.Fragment, of: Ash.Resource

  relationships do
    belongs_to :company, MyApp.Company.Company do
      allow_nil? false
    end

    belongs_to :quick_message, MyApp.Messaging.QuickMessage do
      primary_key? true
      allow_nil? false
    end

    belongs_to :quick_responses, MyApp.Messaging.QuickResponses do
      primary_key? true
      allow_nil? false
    end
  end
end
```

Running:

```
mix credo --strict --checks AshCredo.Check.Warning.MissingPrimaryKey
```

Output:

```
[W] ↗ Resource is missing a primary key.
      lib/my_app/messaging/resources/quick_message_response.ex:3
        #(MyApp.Messaging.QuickMessageResponse)
```

`Ash.Resource.Info.primary_key(MyApp.Messaging.QuickMessageResponse)` returns `[:quick_message_id, :quick_responses_id]` at runtime — the composite PK is correctly assembled from the fragment.

## Root cause

`AshCredo.Check.Warning.MissingPrimaryKey.check_for_primary_key/2` walks the AST of the resource source file only. The `fragments: [...]` option is not followed, so PK-bearing entities declared in fragment modules are invisible to the check.

## Suggested fixes (any one would resolve)

1. **Follow `fragments:`.** When `use Ash.Resource, fragments: [...]` lists modules, also AST-walk those modules' `relationships`/`attributes` blocks for PK entities. This stays in the static-analysis mode the check already uses, but extends it to its natural scope.
2. **Use a compiled-project check.** Switch to `Ash.Resource.Info.primary_key/1` like `UnknownAction` and `MissingCodeInterface` do. This fixes fragments AND extension-provided PKs (see companion issue) in one stroke.
3. **Provide a check-level escape hatch** such as `excluded_when_fragments_present?: true`, but this is less precise than option 1 or 2.

**Comment by leonqadirie** [comment]
Thank you for the three issues!

Looking deeper into it I'd either use a compiled-source check OR just drop the whole `MissingPrimaryKey` check from `AshCredo` completely, as `Ash` actually has a built-in compile-time **warning** for that. Do you see a strong reason to keep the check at all?

**Comment by jeroen11dijk** [comment]
I dont think I had ever seen the warning, but I doubt that the check adds much, especially in combination with all other checks such as having a code interface.

**Comment by leonqadirie** [comment]
Yeah me neither, hence I had added it, should have checked Ash's source... I'll ponder it a bit, currently leaning towards just removing the check, though.

Thank you for the nudge :)
